src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,198 @@
+        require(delegate_  != address(0), "D3MCompoundPlan/invalid-delegate");
+
+        cDai     = CErc20Like(cDai_);
+        tack     = InterestRateModelLike(rateModel_);
+        delegate = delegate_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MCompoundPlan/not-authorized");
+        _;
+    }
+
+    // --- Math ---
+    uint256 internal constant WAD = 10 ** 18;
+    function _wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * y) / WAD;
+    }
+    function _wdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * WAD) / y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "barb") {
+            require(data <= MAX_BORROW_RATE, "D3MCompoundPlan/barb-too-high");
+            barb = data;
+        } else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+    function file(bytes32 what, address data) external auth {
+        if (what == "tack") tack = InterestRateModelLike(data);
+        else if (what == "delegate") delegate = data;
+        else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = barb;
+        if (targetInterestRate == 0) return 0; // De-activated
+
+        uint256 borrows = cDai.totalBorrows();
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, borrows);
+        uint256 totalPoolSize = cDai.getCash() + borrows - cDai.totalReserves();
+
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return currentAssets + targetTotalPoolSize - totalPoolSize;
+        } else {
+            // Decrease debt
+            unchecked {
+                uint256 decrease = totalPoolSize - targetTotalPoolSize;
+                if (currentAssets >= decrease) {
+                    return currentAssets - decrease;
+                } else {
+                    return 0;
+                }
+            }
+        }
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 borrows) internal view returns (uint256) {
+        uint256 kink                   = tack.kink();
+        uint256 multiplierPerBlock     = tack.multiplierPerBlock();
+        uint256 baseRatePerBlock       = tack.baseRatePerBlock();
+        uint256 jumpMultiplierPerBlock = tack.jumpMultiplierPerBlock();
+
+        uint256 normalRate = _wmul(kink, multiplierPerBlock) + baseRatePerBlock;
```

Comment:
I think it'd be better called as `maxNormalRate`

Reply:
This is actually a Compound term:
https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L102
Tried to stick with their vocabulary.

Reply:
I see this is how is called in Compound, but at least confused me for a while...

Reply:
Sorry my last comment was without reloading and hadn't seen yours. Yeah I saw the same after. I guess it is fine to keep the same naming, maybe just a comment to clarify would be fine.

Reply:
Added a comment at `c9167a684`